pkg/util/openstack/loadbalancer.go

```diff
@@ -169,15 +170,52 @@ func CreateListener(client *gophercloud.ServiceClient, lbID uint64, opts listene
 
 // GetLoadbalancerByName get the load balancer which is in valid status by the given name.
 func GetLoadBalancersByName(client *gophercloud.ServiceClient, name string) ([]loadbalancers.LoadBalancer, error) {
-	var validLBs []loadbalancers.LoadBalancer
+	var allLoadbalancers, validLBs []loadbalancers.LoadBalancer
+	var listOpts loadbalancerv2.ListOpts
+	var oldMarker string
+
+	// Query all the LBs by using the markers.
+	// Set last LB ID as marker.
+	for {
+		var tempLbs []loadbalancers.LoadBalancer
+		var err error
+
+		err = loadbalancers.List(client, listOpts).EachPage(func(page pagination.Page) (bool, error) {
+			tempLbs, err = loadbalancers.ExtractLoadBalancers(page)
+			if err != nil {
+				return false, err
+			}
 
-	allPages, err := loadbalancers.List(client, loadbalancers.ListOpts{}).AllPages()
-	if err != nil {
-		return nil, err
-	}
-	allLoadbalancers, err := loadbalancers.ExtractLoadBalancers(allPages)
-	if err != nil {
-		return nil, err
+			return false, nil
+		})
+		if err != nil {
+			return nil, err
+		}
+
+		if len(tempLbs) > 0 {
+			marker := strconv.FormatUint(tempLbs[len(tempLbs)-1].ID, 10)
+			if marker == oldMarker {
+				break
+			}
+
+			// First time in the loop add all the LBs (or) if oldMarker is not
+			// same as the first LB ID in newly retrieved LBs then no need to
+			// remove it.
+			// Else check if oldMarker is same as the first LB ID in newly retrieved LBs then
+			// we can remove it.
+			tempId := strconv.FormatUint(tempLbs[0].ID, 10)
```

Comment:
rename as firstId ? might help for somebody reading this code easier ?

Reply:
Refactored the code.. Please review once..